File: C++/src/algorithms/bubble_sort/bubble_sort.cpp
#include "bubble_sort.h"
#include <array>

void bubble_sort(int arr[])
{
    for (int i = 0; i < 5; i++)
    {
        for (int j = 0; j < 5 - 1 - i; j++)
        {
            if (arr[j] > arr[j + 1])
            {
                int tmp = arr[j + 1];
                arr[j + 1] = arr[j];
                arr[j] = tmp;
            }
        }
    }
}

File: CSharp/Algorithms/BubbleSort/BubbleSort.cs
namespace CSharp.Algorithms.BubbleSort;

public class BubbleSort
{
    public static int[] Sort(int[] arr)
    {
        int[] res = arr;

        for (int i = 0; i < arr.Length; i++)
        {
            for (int j = 0; j < arr.Length - i - 1; j++)
            {
                if (res[j] > res[j + 1])
                {
                    int next = res[j + 1];
                    res[j + 1] = res[j];
                    res[j] = next;
                }
            }
        }

        return res;
    }
}
